components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationManagerImpl.java

```diff
@@ -929,20 +946,39 @@ private String getPaginatedFilter(String paginatedFilter, String after, String b
             if (StringUtils.isNotBlank(before)) {
                 String decodedString = new String(Base64.getDecoder().decode(before), StandardCharsets.UTF_8);
                 Timestamp.valueOf(decodedString);
-                paginatedFilter += StringUtils.isNotBlank(paginatedFilter) ? " and before gt " + decodedString :
-                        "before gt " + decodedString;
+                paginatedFilter += StringUtils.isNotBlank(paginatedFilter) ? " and before lt " + decodedString :
+                        "before lt " + decodedString;
             } else if (StringUtils.isNotBlank(after)) {
                 String decodedString = new String(Base64.getDecoder().decode(after), StandardCharsets.UTF_8);
                 Timestamp.valueOf(decodedString);
-                paginatedFilter += StringUtils.isNotBlank(paginatedFilter) ? " and after lt " + decodedString :
-                        "after lt " + decodedString;
+                paginatedFilter += StringUtils.isNotBlank(paginatedFilter) ? " and after gt " + decodedString :
+                        "after gt " + decodedString;
             }
         } catch (IllegalArgumentException e) {
             throw handleClientException(ERROR_CODE_INVALID_CURSOR_FOR_PAGINATION);
         }
         return paginatedFilter;
     }
 
+    private String getPaginatedFilterForMetaAttributes(String paginatedFilter, String after, String before)
+            throws OrganizationManagementClientException {
+
+        try {
+            if (StringUtils.isNotBlank(before)) {
+                String decodedString = new String(Base64.getDecoder().decode(before), StandardCharsets.UTF_8);
+                paginatedFilter += StringUtils.isNotBlank(paginatedFilter) ? " and attributeBefore lt "
+                        + decodedString : "attributeBefore lt " + decodedString;
+            } else if (StringUtils.isNotBlank(after)) {
+                String decodedString = new String(Base64.getDecoder().decode(after), StandardCharsets.UTF_8);
+                paginatedFilter += StringUtils.isNotBlank(paginatedFilter) ? " and attributeAfter gt "
+                        + decodedString : "attributeAfter gt " + decodedString;
```

Comment:
Instead of `before,  after` why did you pick `attributeBefore`, `attributeAfter`?

Reply:
This was used for mapping the filters to the column name in [here](https://github.com/wso2/identity-organization-management-core/pull/144/files#diff-c3d309ce0ff74c9adbf4cb6a6419e35b4834ee5af2dc0f02dd5e081c8d39aab3R995).

The attribute map maps the before/after to the VIEW_CREATED_TIME_COLUMN [here](https://github.com/wso2/identity-organization-management-core/pull/144/files#diff-ae3ba072650382e9c7b02b03bc5cb1471381a2d7a8c1f9e1bc0d3d547dd2652fL159).

Hence the new filter name attributeBefore/attributeAfter was used to map to the VIEW_ATTR_KEY_COLUMN.

However, as discussed offline, we decided to handle this filter mapping logic by using the `attributeUsedForCursor` parameter in the DAO implementation [here](https://github.com/wso2/identity-organization-management-core/pull/144/files#diff-c3d309ce0ff74c9adbf4cb6a6419e35b4834ee5af2dc0f02dd5e081c8d39aab3R999).